components/resource_metering/src/utils.rs

```diff
@@ -3,6 +3,7 @@
 #![allow(dead_code)]
 
 use collections::HashSet;
+use std::thread;
```

Comment:
It causes a warning if the build platform is linux. I suggest you write `std::thread::current().id().as_u64().get() as usize` in the `thread_id` function.

Reply:
done